### ISSUE TITLE
UK Roundabout Roads - Pedestrians still crossing to center of road

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#786]
There are several roundabout roads in the UK Roads project asset collections, for example this one:

https://steamcommunity.com/workshop/filedetails/?id=1936459427

They only have pedestrian lane on one edge (what should be outer edge) of the road, the idea being that pedestrians will not cross to the center of the roundabout.

A steam user just reported that pedestrians are still crossing to center of roundabout even with crossings disabled in junction restrictions tool.

> **Painful Agenda:** Hi I am using [UK Roads](https://steamcommunity.com/workshop/filedetails/?id=1936459427) and when I switch off pedistrians on roundabouts they still walk and cross at the junctions in the roundabout even if I provide footpaths and disable all pedestrians

This is likely a pathfinder glitch, it seems to treat edges of roads as pedestrian lanes even if there is no pedestrian lane. I've noticed this also on the Rural roads from same asset creator which from memory don't have pedestrian lanes but pedestrians can still walk along them (I could be wrong on that).

It could also potentially be an issue with RHT vs. LHT. The roads are designed primarily for LHT so it's possible there is a glitch when used in RHT cities?

Suggestion: A tool to highlight pedestrian lanes in camera view would be useful starting point.

**Comment by kianzarrin** [comment]
Pedestrians can cross highway roads too.

**Comment by kianzarrin** [comment]
Pedestrians can cross the street but If there is no pedestrian lane on the other side I see no reason why pedestrians should go there. I suspect he has got his roundabout backward. or maybe its RHT LHT.

**Comment by kianzarrin** [comment]
Without proper picture I think this issue will remain open for ever!

**Comment by aubergine10** [comment]
I think we will need some visualisation for ped lanes before any screenshots would be meaningful. Without that we just have no idea where the ped lanes are.

The road editor is extremely badly designed, so even asset authors are not entirely sure how everything is working. Depending on factors such as LHT vs. RHT, and even the direciton in which the road is placed on the map - clockwise vs. anticlockwise, ... all these things affect it.

Without visualisation of ped lanes, we will always be left guessing.

**Comment by kianzarrin** [comment]
I'll just create a mod to do it:)

**Comment by aubergine10** [comment]
Tested the roads with Kain's new mod, and they only have ped lane on one side. I'll continue examining to see if I can a) reliably replicate the issue and b) work out if there's any other wierd stuff happening with the lanes.

**Comment by aubergine10** [comment]
Just noticed I never added the img of the road with single ped lane... this is from UK Roads Revived roundabout roads.

![image](https://user-images.githubusercontent.com/1386719/143790561-2d2db11b-a86f-433f-99d5-a5a5aaac195a.png)

**Comment by krzychu124** [comment]
I remember that @chameleon-tbn reported similar issue some time ago, also with pedestrian lane on one side 🤔

**Comment by aubergine10** [comment]
In OP of this issue I also mentioned that cims were walking on edges of rural roads, however upon inspection with Kian's mod (which wasn't available at the time) it seems those roads do actually define ped lanes on both sides.

![image](https://user-images.githubusercontent.com/1386719/143790914-628bb182-0a72-4dc1-ba14-ed70512a16aa.png)

I'll edit OP to strikeout the misleading info.

**Comment by chameleon-tbn** [comment]
> I remember that @chameleon-tbn reported similar issue some time ago, also with pedestrian lane on one side 🤔

That's correct. It can't be be 'cause of LHT/RHT, as the Roads I've used where designed for RHT and have the Path Just on outer side... As i have unsubbed all Roads with this issue idk exactly which ones i used. But imho this ones where effected: https://steamcommunity.com/sharedfiles/filedetails/?id=2238666416

**Comment by kianzarrin** [comment]
does this mean we can close this?
